### PR TITLE
DEMO-118 : follow artist& unfollow artist & user schema fix

### PR DESCRIPTION
**Pull request by sseerrnn**
###  Change follow system infrastructure
-  using user._id to be a key for follow system
-  store only user following id

### CODE REVIEW

**Comment by sseerrnn** [comment]
### user schema
- _id: Types.ObjectId 
-   username: string
-   password: string
-   email: string
-   registrationDate: Date
-   accountNumber: number
-   bankName: string
-   roles: Role[]
-   following: Types.ObjectId[]
-   profileImage: string

File: src/users/users.controller.ts [hunk 1]
```diff
@@ -244,4 +244,22 @@ export class UsersController {
     const user = await this.usersService.findOneByEmail(req.user.email);
     return user.roles;
   }
+
+  @ApiBearerAuth()
+  @ApiParam({ name: 'followeename' })
+  @Put('follow/:followeename')
```
Comment (pol-alones) [inline src/users/users.controller.ts]: Should be `followeeName`

File: src/users/users.controller.ts [hunk 2]
```diff
@@ -244,4 +244,22 @@ export class UsersController {
...
+  @HttpCode(HttpStatus.OK)
+  async followAritist(@Req() req, @Param() params) {
+    await this.usersService.followArtist(req.user.email, params.followeename);
+    return { message: 'success to follow user', success: true };
+  }
+
+  @ApiBearerAuth()
+  @ApiParam({ name: 'followeename' })
+  @Put('unfollow/:followeename')
```
Comment (pol-alones) [inline src/users/users.controller.ts]: `followeeName`

File: src/users/users.controller.ts [hunk 3]
```diff
@@ -244,4 +244,22 @@ export class UsersController {
...
+  @HttpCode(HttpStatus.OK)
+  async unfollowAritist(@Req() req, @Param() params) {
```
Comment (pol-alones) [inline src/users/users.controller.ts]: Artist

File: src/users/users.controller.ts [hunk 4]
```diff
@@ -244,4 +244,22 @@ export class UsersController {
     const user = await this.usersService.findOneByEmail(req.user.email);
     return user.roles;
   }
+
+  @ApiBearerAuth()
+  @ApiParam({ name: 'followeename' })
+  @Put('follow/:followeename')
+  @HttpCode(HttpStatus.OK)
+  async followAritist(@Req() req, @Param() params) {
```
Comment (pol-alones) [inline src/users/users.controller.ts]: Artist

File: src/users/users.service.ts [hunk 5]
```diff
@@ -126,4 +126,52 @@ export class UsersService {
 
     return await this.update(user._id.toString(), user);
   }
+
+  async followArtist(
+    followerEmail: string,
+    followeename: string,
```
Comment (pol-alones) [inline src/users/users.service.ts]: `followeeName`

File: src/users/users.service.ts [hunk 6]
```diff
@@ -126,4 +126,52 @@ export class UsersService {
...
+      throw new ConflictException(`User already follows ${followeename}`);
+    }
+    follower.following.push(followee._id);
+
+    await this.update(follower._id.toString(), follower);
+  }
+
+  async unfollowArtist(
+    followerEmail: string,
+    followeename: string,
```
Comment (pol-alones) [inline src/users/users.service.ts]: followeeName

**Review by pol-alones — APPROVED** [review]
Just fix some typo.

**Pull request merged**